### DISCUSSION TITLE
Deploying winget to new installations of Windows 10

### DISCUSSION THREAD

**Discussion by mariobrostech** [microsoft/winget-cli#1195] [Q&A] [answered]
Hello,

I've been trying to figure out how to install winget automatically on all of my new installations of Windows 10 since I'm expecting to do a round of deployments soon. I'm pretty new to UWP app deployment and would appreciate any guidance you may have about this.

I've come across 3 Microsoft Docs articles that are extremely helpful, and I encourage you to take a look at them so you have a better idea of what I'm trying to accomplish with this:

1. [How to install and update Desktop framework packages](https://docs.microsoft.com/en-us/troubleshoot/cpp/c-runtime-packages-desktop-bridge#how-to-install-and-update-desktop-framework-packages). Provides the VC++ v14 Desktop Framework Package dependency needed to deploy winget on Windows 10, Version 1809.
2. [Preinstall a Microsoft Store-signed app with DISM](https://docs.microsoft.com/en-us/windows-hardware/manufacture/desktop/preinstall-apps-using-dism#preinstall-a-microsoft-store-signed-app-with-dism). Guides you through mounting an `install.wim` file and lists sample commands for adding the winget appxbundle to the offline image.
3. [App package servicing options](https://docs.microsoft.com/en-us/windows-hardware/manufacture/desktop/dism-app-package--appx-or-appxbundle--servicing-command-line-options#app-package-servicing-options). Goes into more detail with the appxbundle sample commands presented in the previous article.

While these articles have been very helpful, they don't address one problem I haven't been able to solve. What am I supposed to use as my license XML file? Here is what I have I have so far:

`dism /Image:C:\Mount /Add-ProvisionedAppxPackage /PackagePath:C:\Users\Deployment\Downloads\Microsoft.DesktopAppInstaller_8wekyb3d8bbwe.appxbundle /DependencyPackagePath:C:\Users\Deployment\Downloads\Microsoft.VCLibs.x64.14.00.Desktop.appx /LicensePath=C:\Users\Deployment\Downloads\???.xml /region=all"`

If anyone could tell me where I could find a copy of the license XML file or could guide me in creating one, I'd really appreciate it! Also, if you have any additional tips on deploying winget using an offline installation of Windows 10, please feel free to share.

Thanks in advance!

**Comment by denelon** [comment, marked as answer]
@mariobrostech we haven't published the license file for "offline" deployment yet. I'll see about adding it in the next release. We're preparing for one now.

**Reply by mariobrostech** [reply]
Thanks, I'd really appreciate that! Would you mind also clarifying if that license file would work when installing winget in conjunction VC++ v14 Desktop Framework Package dependency, or could it only be used for the winget package itself?

**Reply by denelon** [reply]
Each MSIX package requires it's own license file for offline installation.

**Reply by mariobrostech** [reply]
@denelon Thanks for the help! Do you have an approximate timeline for when the license file will roll out so I can plan ahead for my deployments?

**Reply by denelon** [reply]
I'm hoping to get it out ASAP.

**Reply by mariobrostech** [reply]
Great! Please let me know when this rolls out :)

**Reply by denelon** [reply]
https://github.com/microsoft/winget-cli/releases/tag/v1.0.11692

**Reply by mariobrostech** [reply]
Thank you!
